File: LocadoraAPI/LocadoraAPI/Models/Locacao.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace LocadoraAPI.Models
{
    public partial class Locacao
    {
        public int IdLocacao { get; set; }
        public int IdVeiculo { get; set; }
        public int IdCliente { get; set; }
        public DateTime DtLocacao { get; set; }
        public DateTime DtDevolucao { get; set; }

        [JsonIgnore]
        public Cliente Cliente { get; set; }
        [JsonIgnore]
        public Veiculo Veiculo { get; set; }
    }
}

File: Locadora/src/app/cadastros/model/locacao.ts
import { Veiculo } from "./veiculo";
import { Cliente } from "./cliente";

export class Locacao {
    idLocacao: number;
    idVeiculo: number;
    idCliente: number;
    dtLocacao: Date;
    dtDevolucao: Date;
    cliente: Cliente;
    veiculo: Veiculo;
    
}
